lib/models/s3-event.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+const S3Event = function (eventData, reqHeaders) {
+  var principal = [];
+  try {
```

Comment:
Instead of using a `try..catch` block and retrieving an element from the `principal` array later in this function, can you use an `if..else` block or ternary and assign a precise `principalId` value here? The current way this is written increases the complexity of understanding your intent.

Reply:
It's been month and I forgot why I did it that way before.  I have removed that block anyways as I don't see any principalId information in request headers.